qlib/kernel/fs/host/util.rs

```diff
@@ -286,7 +293,10 @@ pub fn TryOpenAt(dirfd: i32, name: &str, skiprw: bool) -> Result<(i32, bool, Lib
         fstat: &fstat,
         writeable: false,
     };
+    #[cfg(not(feature = "cc"))]
     let cstr = CString::New(&name);
+    #[cfg(feature = "cc")]
+    let cstr = SharedString::New(&name);
```

Comment:
I see this pattern a lot. IMHO this doesn't scale well: whenever you need to share a string with the host you need to repeat this pattern, and those who doesn't work on cc would easily ignore it and breaks the cc code.

I would suggest using "SharedString" wherever it's used for guest/host communication. And implement them differently for cc/non-cc:
for cc: it's allocated on the shared heap, for non-cc, it's the same as CString.

Also in general, I think it should be explicit that an object is for sharing, be it cc or not.

Reply:
I agree to use "SharedString" wherever it's used for guest/host communication.  I can move the cfg here and add it into ShareString for cc/non-cc. Then ShareString will be a common mod but not cc specific.

Reply:
> Then ShareString will be a common mod but not cc specific.

It doesn't need to be CC specific. This is a general concept that "xyz" is for guest/host communication. Also regarding naming, CString and String means different things I think. CString is a null-terminated char array, while String is a more complicated object on the heap.

Anyways, I'm not saying you need to do the changes _now_ because we are looking at a systematic problem regarding host/guest comminication. sooner or later we need to deal with the interface. Passing a raw pointer to the host is very unsafe.